src/NuGet.Core/NuGet.Packaging/PackageCreation/Authoring/ManifestReader.cs

```diff
@@ -86,7 +86,9 @@ private static void ReadMetadataValue(ManifestMetadata manifestMetadata, XElemen
                         manifestMetadata.Id = value;
                         break;
                     case "version":
-                        manifestMetadata.Version = NuGetVersion.Parse(value);
+                        NuGetVersion version;
+                        NuGetVersion.TryParse(value, out version);
+                        manifestMetadata.Version = version ?? new NuGetVersion(0, 0, 0);
```

Comment:
What if we just allowed it be null instead of a default? 
Do we ever need to differentiate between that and a 0,0,0?

Would we detect that no version has been set somewhere different?

Another idea would be, what if we didn't set a version element in the nuspec.

Reply:
This version is used to create the nupkg name, so in this case an empty version in the nuspec will create the package as `package.0.0.0.nupkg` so most likely there is a way to detect that no version has been set later in the code.

> Another idea would be, what if we didn't set a version element in the nuspec.
I'm not sure what would this cause, for me it looks like version is very important

Reply:
> I'm not sure what would this cause, for me it looks like version is very important

I was thinking as more of a test case actually. What would happen if there was no version specified in the nuspec when we attempt to pack and just pass it on the CLI. 

That feels like the closest analog to this test case.

Version is a mandatory element, so I'm just worried about the potential fall out of setting a default version.

How are token replacements done? Can we parse the CLI version and validate it before the we read the manifest?

Reply:
> I was thinking as more of a test case actually. What would happen if there was no version specified in the nuspec when we attempt to pack and just pass it on the CLI.

Currently, if you don't have the version tag it will fail saying that the file is incorrect, if it is empty, it will fail because we try to parse that empty string and fail if the parsing returns null.

> Can we parse the CLI version and validate it before the we read the manifest?

We do validate the CLI version before reading the manifest, and after reading the manifest we check if we specified a CLI version and replace it with it. 

So I think after this conversation I think the best solution is:

1. Continue validating the CLI version
2. Don't immediately break when there is an empty Version in the nuspec
3. When we would do the "replacement", if CLI version is invalid and there is no version in nuspec break. Otherwise continue as how it is right now. (if CLI version is valid it would override the .nuspec version)

Reply:
I was thinking, what if we validate once? 
Can we inject the CLI version into the nuspec before reading the manifest, or is that not possible?

Reply:
Sorry for the late reply, I think I was able to address this concern, I'm validating the CLI version and updating the manifest version if it's a valid version.